### PR TITLE
Launch merging api new structure

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/core/configs/MergeStrategyConfig.java [hunk 1]
```diff
@@ -21,22 +21,40 @@
...
+	private final TestItemRepository testItemRepository;
+
+	@Autowired
+	public MergeStrategyConfig(TestItemRepository testItemRepository) {
+		this.testItemRepository = testItemRepository;
+	}
+
+	@Bean
+	public Map<MergeStrategyType, MergeStrategy> mergeStrategyMapping() {
+		Map<MergeStrategyType, MergeStrategy> mapping = new HashMap<>();
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/configs/MergeStrategyConfig.java]: singleton map or at least hashmap with size equal to 1

File: src/main/java/com/epam/ta/reportportal/core/item/impl/merge/strategy/AbstractSuiteMergeStrategy.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2016 EPAM Systems
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/item/impl/merge/strategy/AbstractSuiteMergeStrategy.java]: apache 2.0

File: src/main/java/com/epam/ta/reportportal/core/item/impl/merge/strategy/AbstractSuiteMergeStrategy.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	AbstractSuiteMergeStrategy(TestItemRepository testItemRepository) {
+		this.testItemRepository = testItemRepository;
+	}
+
+	public abstract boolean isTestItemAcceptableToMerge(TestItem item);
+
+	@Override
+	public TestItem mergeTestItems(TestItem itemTarget, List<TestItem> items) {
+		TestItem result = items.stream().reduce(itemTarget, this::moveAllChildTestItems);
+		TestItem result1 = items.stream().reduce(itemTarget, (prev, curr) -> moveAllChildTestItems(prev, curr));
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/item/impl/merge/strategy/AbstractSuiteMergeStrategy.java]: is result1 used somewhere?

File: src/main/java/com/epam/ta/reportportal/core/item/impl/merge/strategy/AbstractSuiteMergeStrategy.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	/**
+	 * Collects tags, parameters and descriptions from source and add them to target. Same tags
+	 * are added only once. Updates start and end times of target. Updates item identifier.
+	 *
+	 * @param target item to be merged
+	 * @param source item to merge
+	 */
+	private void updateTargetItemInfo(TestItem target, TestItem source) {
+		target = updateTime(target, source);
+		source.getTags().removeAll(target.getTags());
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/item/impl/merge/strategy/AbstractSuiteMergeStrategy.java]: check for null, please

File: src/main/java/com/epam/ta/reportportal/core/item/impl/merge/strategy/MergeStrategyType.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2016 EPAM Systems
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/core/item/impl/merge/strategy/MergeStrategyType.java]: apache 2.0 everywhere

File: src/main/java/com/epam/ta/reportportal/core/item/impl/FinishTestItemHandlerImpl.java [hunk 6]
```diff
@@ -161,5 +162,9 @@ private void verifyTestItem(ReportPortalUser user, TestItem testItem, Optional<S
 				"There is no status provided from request and there are no descendants to check statistics for test item id '{}'",
 				testItem.getItemId()
 		));
+
+		expect(finishExecutionRQ.getEndTime(), date -> TO_LOCAL_DATE_TIME.apply(date).isAfter(testItem.getStartTime())).verify(FINISH_TIME_EARLIER_THAN_START_TIME,
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/item/impl/FinishTestItemHandlerImpl.java]: as i remember, we decided to remove this restriction in 5.0v

File: src/main/java/com/epam/ta/reportportal/core/item/impl/GetTestItemHandlerImpl.java [hunk 7]
```diff
@@ -77,21 +82,19 @@ public TestItemResource getTestItem(Long testItemId) {
 	 * .lang.String, java.util.Set, org.springframework.data.domain.Pageable)
 	 */
 	@Override
-	public Iterable<TestItemResource> getTestItems(Queryable filterable, Pageable pageable) {
-//		return itemAssembler.toPagedResources(testItemRepository.findByFilter(filterable, pageable));
-		throw new UnsupportedOperationException("No implementation");
+	public Iterable<TestItemResource> getTestItems(Filter filter, Pageable pageable) {
+		Page<TestItem> testItems = testItemRepository.findByFilter(filter, pageable, TEST_ITEM_FETCH);
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/item/impl/GetTestItemHandlerImpl.java]: can we move TEST_ITEM_FETCH to dao layer?

File: src/main/java/com/epam/ta/reportportal/core/item/impl/merge/strategy/AbstractSuiteMergeStrategy.java (hunk 3 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/item/impl/merge/strategy/AbstractSuiteMergeStrategy.java]: it was a missprint

**Pull request merged**